### PR TITLE
Add stitching to users table

### PR DESCRIPTION
**Pull request by rlh1994**
## Description

This PR adds the ability to do user stitching on the users table. This seems like an odd oversight from the past, as without it the user_id column is based on the first session that user has and is never updated. This allows us to have user stitching across all derived tables and removes the need for users to join to the user mapping table themselves

Also fixed a wrong link in the readme.

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
https://snplow.atlassian.net/browse/PE-5794

## Checklist
- [x] 🎉 I have verified that these changes work locally
- [ ] 💣 Is your change a breaking change?
- [x] 📖 I have updated the CHANGELOG.md

### Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [x] 📓 internal package docs (ymls, macros, readme, if applicable)
- [ ] 📕 I have raised a [Snowplow documentation](https://github.com/snowplow/documentation) PR if applicable (Link here if required)
- [ ] 🙅 no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

image

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
+snowplow-unified 0.2.0 (2024-0X-XX)
 ---------------------------------------
 ## Summary
 
+## Features
+- Adds user stitching to the users table (enabled with `snowplow__session_stitching`)
+
 ## Fixes
 - Consider screen view ID from the screen view context (#14)
+- Fix link to incorrect FAQ in README
```
Comment (agnessnowplow) [inline CHANGELOG:10]: Previous fix needs to be removed

**Review by agnessnowplow — APPROVED** [review]
All good, apart from the changelog but that can be fixed later as well.

File: CHANGELOG (hunk 1 above)
Comment (rlh1994) [reply CHANGELOG:10]: That's not yet released is it?

Comment (agnessnowplow) [reply CHANGELOG:10]: Oh, it looked like it did, ignore it pls.

**Pull request merged**